damus/Views/Events/FollowPack/FollowPackPreview.swift

```diff
@@ -0,0 +1,240 @@
+//
+//  FollowPackPreview.swift
+//  damus
+//
+//  Created by eric on 4/30/25.
+//
+
+import SwiftUI
+import Kingfisher
+
+struct FollowPackUsers: View {
+    let state: DamusState
+    var publicKeys: [Pubkey]
+    
+    var body: some View {
+        HStack(alignment: .center) {
+
+            if !publicKeys.isEmpty {
+                CondensedProfilePicturesView(state: state, pubkeys: publicKeys, maxPictures: 5)
+            }
+
+            let followPackUserCount = publicKeys.count
+            let nounString = pluralizedString(key: "follow_pack_user_count", count: followPackUserCount)
+            let nounText = Text(verbatim: nounString).font(.subheadline).foregroundColor(.gray)
+            Text("\(Text(verbatim: followPackUserCount.formatted()).font(.subheadline.weight(.medium))) \(nounText)", comment: "Sentence composed of 2 variables to describe how many people are in the follow pack. In source English, the first variable is the number of users, and the second variable is 'user' or 'users'.")
+        }
+    }
+}
+
+struct FollowPackBannerImage: View {
+    let state: DamusState
+    let options: EventViewOptions
+    var image: URL? = nil
+    var preview: Bool
+    @State var blur_imgs: Bool
+    
+    func Placeholder(url: URL, preview: Bool) -> some View {
+        Group {
+            if let meta = state.events.lookup_img_metadata(url: url),
+               case .processed(let blurhash) = meta.state {
+                Image(uiImage: blurhash)
+                    .resizable()
+                    .frame(maxWidth: preview ? 350 : UIScreen.main.bounds.width, minHeight: preview ? 150 : 200, maxHeight: preview ? 150 : 200)
+            } else {
+                DamusColors.adaptableWhite
+            }
+        }
+    }
+    
+    func titleImage(url: URL, preview: Bool) -> some View {
+        KFAnimatedImage(url)
+            .callbackQueue(.dispatch(.global(qos:.background)))
+            .backgroundDecode(true)
+            .imageContext(.note, disable_animation: state.settings.disable_animation)
+            .image_fade(duration: 0.25)
+            .cancelOnDisappear(true)
+            .configure { view in
+                view.framePreloadCount = 3
+            }
+            .background {
+                Placeholder(url: url, preview: preview)
+            }
+            .aspectRatio(contentMode: .fill)
+            .frame(maxWidth: preview ? 350 : UIScreen.main.bounds.width, minHeight: preview ? 150 : 200, maxHeight: preview ? 150 : 200)
+            .kfClickable()
+            .cornerRadius(1)
+    }
+    
+    var body: some View {
+        if let url = image {
+            if (self.options.contains(.no_media)) {
+                EmptyView()
+            } else if !blur_imgs || (!blur_imgs && !state.settings.media_previews) {
+                titleImage(url: url, preview: preview)
+            } else if blur_imgs || (blur_imgs && !state.settings.media_previews) {
```

Comment:
I don't think these boolean expressions make sense. The second half of the OR expression in both else ifs will always be false. And the last else if will always be true. I'm not sure how the media previews settings play into this either. Did you mean something like this?

```suggestion
            } else if !blur_imgs {
                titleImage(url: url, preview: preview)
            } else {
```